documentation: adds contributing.md

priley86 opened the pull request:
Adds code contribution and contribution process guidelines for the Patternfly React project.

issue: #4 

cc: @jgiardino @maryclarke @LHinson @bleathem

LHinson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+# Contributing to Patternfly React
```
all instances of "Patternfly" should be updated to "PatternFly"

priley86 replied:
will update

LHinson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+React component contributions should observe the following high level process:
+
+1. Check the [Patternfly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to Patternfly.
+2. Create an `enhancement` issue to indicate you'd like to contribute this component to Patternfly React and notify the Patternfly React maintainer team.
```
you reference the "PatternFly React maintainer team" however it does not provide who this anywhere. As someone reading this and unfamiliar with the team, I wouldn't be sure who to reach out to. What do you think about identifying the team members?

LHinson replied:
"Create an `enhancement` issue to indicate you'd like to contribute this component to Patternfly React and notify the Patternfly React maintainer team." <<< via? how does the team want contacted? the mailing list?

jgiardino replied:
If we're all subscribed to this repo, then will we automatically get notifications when new issues are created? If so, then I ok with just saying "Create an `enhancement` issue to indicate you'd like to contribute this component to PatternFly React."

LHinson replied:
That approach makes sense to me @jgiardino

jgiardino replied:
Updating my suggestion to include more details about the review process:

> Create an enhancement issue to indicate you'd like to contribute this component to PatternFly React. At this time, the PatternFly React design team will review the enhancement issue and verify that the component passes the decision tree. Only components that pass will be accepted into the repo. If the component does not pass, the team will work with you on identifying any sub-components that can be extracted that would be needed by more than one project.

Please review and let me know if this accurately captures what @maryclarke and @wrnkld and Rox will do. I don't want to make any promises. :-) 

Also let me know if this should be broken out somewhere else.

jgiardino replied:
@LHinson Do other repo teams state who their core members are? If so, where do they add this information?

jgiardino commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+
+- `bug` - A bug is a _demonstrable problem_ that is caused by the code in the repository. Please check if the issue has already been reported before reporting a new bug.
+- `duplicate` - A duplicate signifies that another issue encapsulates or restates this issue. Duplicates will be closed and referenced as such. 
+- `ehancement` - An ehancement is a feature request. Please provide as much detail as possible and consider whether your idea fits the scope and aims of this project.
```
Typo here for 'enhancement'

priley86 replied:
good catch. will fix.

jgiardino commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+* Provide an associated `.stories.js` [story](https://getstorybook.io/docs/react-storybook/basics/writing-stories) for your component. Stories should demonstrate as many different UI states for your component as possible. Use Storybook [knobs](https://github.com/storybooks/storybook-addon-knobs) to enable dynamic visualizations of your component's props.
+* Provide a [jest snapshot test](https://facebook.github.io/jest/docs/snapshot-testing.html) to ensure your UI markup does not change unexpectedly.
+* Ensure the component's rendered design and Storybook stories meet [Patternfly design standard](https://github.com/patternfly/patternfly-design). 
+**Note:** If your component does not yet have Patternfly design documentation, please meet with the Patternfly React maintainer team to ensure this gets contributed along with your component. 
```
Similar to Leslie's comment above, this states to contact team members but doesn't describe the process for that. Any suggestions/preferences on how the design team wants to handle this process?

If the necessary reviewers will already receive notifications, we could just change this statement to something like "If your component does not yet have Patternfly design documentation, the PatternFly React design team will first confirm that the pattern passes the [Patternfly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) and then start the process for generating design documentation"

jgiardino commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+
+7. Export Storybook 
+
+Lastly, you'll want to export Storybook in your fork and note the url afterward.
```
can we change this to 
`...and include the Storybook url in your PR.`

jgiardino replied:
Oh, I see you have that below :-)

jgiardino commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+* This repository serves as a UI / presentational component library only. This means we should not be introducing container components which subscribe to state updates or handle data fetching (i.e. redux aware components). Prefer [stateless functional components](http://buildwithreact.com/article/stateless-functional-components)
+when possible and accept [props](https://facebook.github.io/react/docs/components-and-props.html) as UI display parameters.
+* Provide a [single default export](http://exploringjs.com/es6/ch_modules.html#_single-default-export) for exporting your React UI component as an ES6 Module in your component's jsx definition.
+* Provide an associated `.stories.js` [story](https://getstorybook.io/docs/react-storybook/basics/writing-stories) for your component. Stories should demonstrate as many different UI states for your component as possible. Use Storybook [knobs](https://github.com/storybooks/storybook-addon-knobs) to enable dynamic visualizations of your component's props.
```
Should we also include comments about the template that we have defined, such as: 

> Include a link to the design documentation in the [PatternFly Pattern Library[(http://www.patternfly.org/pattern-library/) if it exists. If a PatternFly design does not exist yet, then provide a description that explains when the component would be used and what goal or task it helps to accomplish.

If we do add this info, then maybe we should move some of the storybook details to sub-bullets?

priley86 replied:
Sounds good to me @jgiardino. Will update.

jgiardino commented:
Thanks for doing this, @priley86! I included some comments inline. Do we also want to state somewhere in this document that if someone contributes to this repo, that we would also like them to participate in reviewing PRs from other contributors?

priley86 commented:
Can you provide some language for this @jgiardino? Happy to provide that suggestion...

jgiardino commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,145 @@
+3. Implement the component observing our [react component requirements](#react-component-requirements) 
+and [code contribution guidelines](#code-contribution-guidelines)
+4. Submit your pull request and be sure to reference the `enhancement` issue so that others interested in this component may discuss. 
+5. Test your component. Once the pull request is merged, please test the component can successfully be used as intended downstream.
```
As a final statement (i.e. not a list item), lets also include:

> If you submit a PR, then please also contribute by reviewing other PRs.

jgiardino commented:
Included a suggested statement for reviewing PRs in this section:
## React Component Contribution Process

priley86 commented:
I've updated this with a note about `semantic-releases` should we decide to go with this option (see proposal in #13. Also added a Github `PULL_REQUEST_TEMPLATE`. 

The Travis CI build is fixed in #13 and can be ignored in this PR.

Let me know anymore thoughts here before we take flight! 🚀

priley86 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,129 @@
+
+React component contributions should observe the following high level process:
+
+1. Check the [PatternFly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to PatternFly.
```
+1 will add it.

priley86 replied:
Added :)

priley86 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,129 @@
+
+1. Check the [PatternFly Decision Tree](https://github.com/patternfly/patternfly-design/blob/master/resources/decision-tree/PatternflyDecisionTree.pdf) to see if your component meets the criteria for an addition to PatternFly.
+2. Create an `enhancement` issue to indicate you'd like to contribute this component to PatternFly React. At this time, the PatternFly React design team will review the enhancement issue and verify that the component passes the decision tree. Only components that pass will be accepted into the repo. If the component does not pass, the team will work with you on identifying any sub-components that ...
+3. Implement the component observing our [react component requirements](#react-component-requirements) 
```
Added :)

priley86 commented:
needs rebase ;)

LHinson commented:
Looks good to merge and we can continue to iterate on it! @priley86
